### PR TITLE
Base project architecture done : fixes #15

### LINKED ISSUE #1: Investigate EA plugin infrastructure
- [x] Install EA and investigate the mechanism to create plugins.
- [x] Verify working of DEH-COMMON within the project
- [x] Create and setup solution on `development` branch

### LINKED ISSUE #15: Create base project architecture
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of the DEHP-EASYSML
- [x] I have searched [open](https://github.com/RHEAGROUP/DEH-EASYSML/issues) and [closed](https://github.com/RHEAGROUP/DEH-EASYSML/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description

Create the base project structure integrating :

 - [x] Common library
 - [x] Unit Test Project and Unit Tests
 - [x] Setup as EA AddIn 

### System Configuration

- DEHPCommon version:
- Environment (Operating system, version and so on):
- .NET Framework version:
- Additional information:

### PR DESCRIPTION
**Pull request by antoineatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/DEH-EASYSML/pulls) open
- [x] I have verified that I am following the DEH-EASYSML [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/DEH-EASYSML/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description

Fixes #1 and #15 

- Solution created with the project and the test project
- DEH-Common included
- Can be used as EA AddIn

### CODE REVIEW

**Review by alexatrhea — APPROVED** [review]

File: DEHEASysML/App.cs [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+        /// Registers the types that can be resolved by the <see cref="IContainer" />
+        /// </summary>
+        /// <param name="containerBuilder">The <see cref="ContainerBuilder" /></param>
+        private void RegisterTypes(ContainerBuilder containerBuilder)
+        {
+            containerBuilder.RegisterType<Dispatcher>().As<IDispatcher>().SingleInstance();
+        }
+
+        /// <summary>
+        /// Add a header to the log file
```
Comment (nathanatrhea) [inline DEHEASysML/App.cs:217]: missing param doc

File: DEHEASysML/Properties/launchSettings.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "profiles": {
+    "DEHEASysML": {
+      "commandName": "Executable",
+      "executablePath": "C:\\Program Files (x86)\\Sparx Systems\\EA\\EA.exe"
+    }
+  }
+}
```
Comment (nathanatrhea) [inline DEHEASysML/Properties/launchSettings.json:8]: missing EOF

File: DEHEASysML/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+﻿using System.Reflection;
+using System.Runtime.InteropServices;
+
+
+[assembly: AssemblyTrademark("DEH")]
+// Setting ComVisible to false makes the types in this assembly not visible 
+// to COM components.  If you need to access a type in this assembly from 
+// COM, set the ComVisible attribute to true on that type.
+[assembly: ComVisible(true)]
+[assembly: Guid("D50722AD-BBF1-4FCC-9E93-DEB2A7C6F7B0")]
```
Comment (nathanatrhea) [inline DEHEASysML/Properties/AssemblyInfo.cs:10]: missing EOF

File: DEHEASysML/NLog.config [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+  <rules>
+    <!-- add your logging rules here -->
+    <logger name="*" minlevel="Debug" writeTo="file" />
+    <!--
+Write all events with minimal level of Debug (So Debug, Info, Warn, Error and Fatal, but not Trace) to "f"
+<logger name="*" minlevel="Debug" writeTo="f" />
+-->
+  </rules>
+</nlog>
```
Comment (nathanatrhea) [inline DEHEASysML/NLog.config:46]: missing EOF

File: DEHEASysML/DEHEASysML.csproj [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+		</Reference>
+		<Reference Include="System.Web" />
+	</ItemGroup>
+
+	<ItemGroup>
+		<None Update="NLog.config">
+			<CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+		</None>
+	</ItemGroup>
+</Project>
```
Comment (nathanatrhea) [inline DEHEASysML/DEHEASysML.csproj:42]: missing EOF

**Review by nathanatrhea — CHANGES_REQUESTED** [review]
small cosmetic changes

**Review by nathanatrhea — APPROVED** [review]

**Pull request merged**